### PR TITLE
Better error message when the git repository from a PUT step is incompatible with the Cogito configuration.

### PR DESCRIPTION
**Pull request by marco-m-pix4d**
Finally! :-)

Better reviewed commit-per-commit.

Better error message when the git repository from a PUT step is incompatible with the Cogito configuration.

Old error:

    resource source configuration and git repository are incompatible.
    Git remote: "git@github.com:owner/repo.git"
    Resource config: host: github.com, owner: "smiling", repo: "butterfly". wrong git remote

New error:

    the received git repository is incompatible with the Cogito configuration.

    Git repository configuration (received as 'inputs:' in this PUT step):
          url: git@github.com:owner/repo.git
        owner: owner
         repo: repo

    Cogito SOURCE configuration:
        owner: smiling
         repo: butterfly

Part of PCI-2274

### CODE REVIEW

**Review by odormond — APPROVED** [review]

**Review by Esysc — APPROVED** [review]

File: resource/resource.go [hunk 1]
```diff
@@ -319,8 +319,12 @@ func collectInputDirs(dir string) ([]string, error) {
 	return dirs, nil
 }
 
-// Check if dir is a git repository, of the form owner/repo, with an origin accessed
-// over HTTP, HTTPS or SSH.
+// checkRepoDir validates wether DIR, assumed to be received as input of a put step,
```
Comment (aliculPix4D) [inline resource/resource.go:322]: Typo: `wether` ---> `Whether`

File: resource/resource_test.go [hunk 2]
```diff
@@ -695,6 +695,21 @@ Cogito SOURCE configuration:
 			repoURL:     "foo://bar",
```
Comment (aliculPix4D) [inline resource/resource_test.go:695]: Typo in the commit message title? `repro`?

**Review by aliculPix4D — APPROVED** [review]

File: resource/resource_test.go (hunk 2 above)
Comment (marco-m-pix4d) [reply resource/resource_test.go:695]: No, "repro" is a shorthand for "reproduction", a setup that can reproduce a bug.

File: resource/resource.go (hunk 1 above)
Comment (marco-m-pix4d) [reply resource/resource.go:322]: thanks.

File: resource/resource_test.go (hunk 2 above)
Comment (aliculPix4D) [reply resource/resource_test.go:695]: I thought so, but I was not 100% sure.

**Pull request merged**